### PR TITLE
LL-2069 Fix the input bug with trailing zeros

### PR DESCRIPTION
**Pull request by juan-cortes**
Let's keep this fragile equilibrium we have with the `InputCurrency` component going a little bit longer

itvorks

### Type
Bug Fix

### Context

https://ledgerhq.atlassian.net/browse/LL-2069

### Parts of the app affected / Test plan

Any amount input in the app

### CODE REVIEW

File: src/renderer/components/InputCurrency.js [hunk 1]
```diff
@@ -86,7 +88,7 @@ class InputCurrency extends PureComponent<Props, State> {
   };
 
   componentDidMount() {
-    this.syncInput({ isFocused: false });
+    this.syncInput({ isFocused: !!this.props.autoFocus });
```
Comment (gre) [inline src/renderer/components/InputCurrency.js:91]: are you sure this is the actual fix? I mention during the meeting and i think it's around this line https://github.com/LedgerHQ/ledger-live-common/blob/master/src/currencies/sanitizeValueString.js#L49

I suspect (but i might be wrong) that you bypass the sanitizer that we still have to use and if you don't use sanitizer you can typically do things like `000.12` or `0.123456789`, it should reformat to `0.12345678` for bitcoin.

in other words, to me it's a but in live-common sanitizer. sanitizer is stateless also, and it's a bit annoying if we allow values like pasting 0.0000 but why not after all.

**Review by gre — CHANGES_REQUESTED** [review]
see comment

File: src/renderer/components/InputCurrency.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/components/InputCurrency.js:91]: I am sure this is the actual fix.

I remember you mentioned the `sanitizeValueString` during the sprint planning, but the sense I make of the code is that we, on one hand, have the possibility of autofocusing an input and on the other hand we hardcode the initial `isFocused` value to false. 

This leads to the `needsToBeReformatted` check always being true (unless you blur + focus again) for autofocused inputs, and therefore triggering the reformatting of the value as if we blurred the input. This is only triggered after the value changes, that's why we can go 1.000001 but when we go back to 1.00000 the formatted value triggers the onChange again and changes it to 0.

**Comment by juan-cortes** [comment]
> see comment

I replied 👍

File: src/renderer/components/InputCurrency.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/components/InputCurrency.js:91]: And addressing the part about skipping the sanitizer, it's not the case, with my fix you still can't type _0.12345678**9**_ nor _**00**0.123_. Without initializing the `isFocused` correctly when we are passing `autoFocus` it's treating the input as if it were blurred every time the numeric value of the string changes.

**Review by gre — APPROVED** [review]

**Comment by nabil-brn** [comment]
The fix is working but do you think some modifications are needed due to https://ledgerhq.atlassian.net/browse/LL-2172?atlOrigin=eyJpIjoiZDM5YjE4YjA2YzRmNGQ1OGI1YTNjMjliN2Q0NTVhYzAiLCJwIjoiaiJ9 ?

If the fix for LL-2172 could be done in another PR so this PR can be merged **✓**.

**Review by nabil-brn — APPROVED** [review]
LL-2172 will be fixed in another PR, seen with @juan-cortes 

inputbug

**✓** QA-ed

---
52d76d7 (HEAD, ledger/LL-2069) LL-2069 Fix the input bug with trailing zeros
